lib/streamlit/server/Server.py

```diff
@@ -116,8 +118,34 @@ def start(self, on_started):
 
         LOGGER.debug("Starting server...")
         app = self._create_app()
+
+        def start_listening(call_count=0):
```

Comment:
Pull this into a top-level function. This way `start()` becomes more readable:

```
  def start(self, on_started):
       if self._state != State.INITIAL:
            raise RuntimeError("Server has already been started")

        LOGGER.debug("Starting server...")
        app = self._create_app()
        start_listening(app, etc, etc)

        port = config.get_option("server.port")
        LOGGER.debug("Server started on port %s", port)

        self._ioloop.spawn_callback(self._loop_coroutine, on_started)
```

Reply:
done